### PR TITLE
Decouple `SeasonalMovingAverageModel` from `PerSegmentModelMixin`

### LINKED ISSUE #1116: Decouple `SeasonalMovingAverageModel` from `PerSegmentModelMixin`
### 🚀 Feature Request

We want to uncouple `SeasonalMovingAverageModel` from `PerSegmentModelMixin`. Doing this, we can make `SeasonalMovingAverageModel` predict on any new segments because it doesn't require fit to work.

It also can lead to better performance.

### Proposal

* Implement `SeasonalMovingAverageModel` without using `PerSegmentModelMixin` and `NonPredictionIntervalContextRequiredModelMixin`. It should be inherited only from `NonPredictionIntervalContextRequiredAbstractModel`.
* Method `get_model` should return `self`.
* Try to vectorize forecasting logic.

### Test cases

1. Make sure current tests pass. 
2. Adapt tests on `_SeasonalMovingAverageModel`.
3. Fix `test_inference` tests to work on new segments.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #1116.

## Closing issues
Closes #1116.

### CODE REVIEW

File: etna/models/seasonal_ma.py [hunk 1]
```diff
@@ -28,165 +26,155 @@ def __init__(self, window: int = 5, seasonality: int = 7):
...
+
         if len(df) < expected_length:
             raise ValueError(
                 "Given context isn't big enough, try to decrease context_size, prediction_size of increase length of given dataframe!"
             )
 
-        history = df["target"][-expected_length:-prediction_size]
-        if np.any(history.isnull()):
+    def _forecast_df(self, df: pd.DataFrame, prediction_size: int) -> pd.DataFrame:
+        self._validate_prediction_size(df=df, prediction_size=prediction_size)
```
Comment (alex-hse-repository) [inline etna/models/seasonal_ma.py:84]: Add docstring here

File: etna/models/seasonal_ma.py [hunk 2]
```diff
@@ -28,165 +26,155 @@ def __init__(self, window: int = 5, seasonality: int = 7):
...
+        expected_length = prediction_size + self.context_size
+
         if len(df) < expected_length:
             raise ValueError(
                 "Given context isn't big enough, try to decrease context_size, prediction_size of increase length of given dataframe!"
             )
 
-        history = df["target"][-expected_length:-prediction_size]
-        if np.any(history.isnull()):
+    def _forecast_df(self, df: pd.DataFrame, prediction_size: int) -> pd.DataFrame:
```
Comment (alex-hse-repository) [inline etna/models/seasonal_ma.py:83]: May be just `_forecast`? + the same for predict

File: tests/test_models/test_simple_models.py [hunk 3]
```diff
@@ -558,6 +556,21 @@ def test_get_model_after_training(example_tsds, etna_model_class, expected_class
...
 
+@pytest.mark.parametrize(
+    "etna_model_class",
+    (
+        NaiveModel,
+        SeasonalMovingAverageModel,
+        MovingAverageModel,
+    ),
+)
+def test_get_model(example_tsds, etna_model_class):
```
Comment (alex-hse-repository) [inline tests/test_models/test_simple_models.py:567]: Why can't we just improv the previous tests?

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: tests/test_models/test_simple_models.py (hunk 3 above)
Comment (Mr-Geekman) [reply tests/test_models/test_simple_models.py:567]: I think that in #1117 we will just merge this two.

File: etna/models/seasonal_ma.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/models/seasonal_ma.py:84]: For which method? `_forecast` or `_validate_prediction_size`?

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**